### ISSUE TITLE
Signing key expired?

### ISSUE THREAD

**Issue by kipe** [dmacvicar/terraform-provider-libvirt#1035]
### Linux distribution

``` Ubuntu 23.04 ```

### Terraform version

```sh
> terraform -v
Terraform v1.6.0
on linux_amd64
```
___

## Checklist

- [ ] Is your issue/contribution related with enabling some setting/option exposed by libvirt that the plugin does not yet support, or requires changing/extending the provider terraform schema?

  - [ ] Make sure you explain why this option is important to you, why it should be important to everyone. Describe your use-case with detail and provide examples where possible.
  - [ ] If it is a very special case, consider using the _XSLT_ support in the provider to tweak the definition instead of opening an issue
  - [ ] Maintainers do not have expertise in every libvirt setting, so please, describe the feature and how it is used. Link to the appropriate documentation

- [x] Is it a bug or something that does not work as expected? Please make sure you fill the version information below:

## Description of Issue/Question

The provider cannot be installed, as the key used to sign it seems to have expired.

### Setup

main.tf
```terraform
terraform {
  required_providers {
    libvirt = {
      source  = "dmacvicar/libvirt"
      version = "0.7.1"
    }
  }
}
```

### Steps to Reproduce Issue

Just trying to initialize a new main.tf with the provider should be enough.
```sh
> terraform init

Initializing the backend...

Initializing provider plugins...
- Finding dmacvicar/libvirt versions matching "0.7.1"...
- Installing dmacvicar/libvirt v0.7.1...
╷
│ Error: Failed to install provider
│ 
│ Error while installing dmacvicar/libvirt v0.7.1: error checking signature: openpgp: key expired
╵
```

**Comment by rabin-io** [comment]
Same issue here, 

Is there an option to manually install the file and pypass the check?

**Comment by southsidedean** [comment]
Yeah, same issue here.

<img width="1102" alt="Screenshot 2023-10-05 at 4 08 04 PM" src="https://github.com/dmacvicar/terraform-provider-libvirt/assets/88509143/63972e3e-4159-4ebb-a533-081d58e9d229">

**Comment by jrocha25** [comment]
Same issue with me

```
Terraform v1.6.0
on windows_amd64
```

**Comment by southsidedean** [comment]
For now, if you roll back to 1.5.7 you should be good. Worked for me.

**Comment by michaelbeaumont** [comment]
Just FYI this is caused by what seems to be [an unintentional breaking changes in `v1.6.0`](https://github.com/hashicorp/terraform/issues/33984#issuecomment-1749126689)

**Comment by kubealex** [comment]
this is related to the breaking change in the recent release: https://github.com/hashicorp/terraform/issues/33984

**Comment by dmacvicar** [comment]
I am looking into it, and publish a new release.

**Comment by dmacvicar** [comment]
Can anyone check if v0.7.2 is working correctly?

**Comment by gwritz** [comment]
Not working for me:
`$ terraform init

Initializing the backend...
Initializing modules...
Downloading registry.terraform.io/MonolithProjects/vm/libvirt 1.10.0 for controlplane...
- controlplane in .terraform/modules/controlplane
Downloading registry.terraform.io/MonolithProjects/vm/libvirt 1.10.0 for worker...
- worker in .terraform/modules/worker

Initializing provider plugins...
- Finding dmacvicar/libvirt versions matching ">= 0.7.0"...
- Installing dmacvicar/libvirt v0.7.2...
╷
│ Error: Failed to install provider
│
│ Error while installing dmacvicar/libvirt v0.7.2: error checking signature: openpgp: key expired
╵
`

**Comment by dmacvicar** [comment]
I forgot to extend the subkey. I am not sure if there is a way to make the registry take the signature without me rebuilding the release.

**Comment by dmacvicar** [comment]
I have rebuild again. Now the registry complains about the hash. It seems the registry is caching the old build :-(

```
Initializing provider plugins...
- Finding dmacvicar/libvirt versions matching "0.7.2"...
- Installing dmacvicar/libvirt v0.7.2...
╷
│ Error: Failed to install provider
│
│ Error while installing dmacvicar/libvirt v0.7.2: checksum list has unexpected SHA-256 hash
│ 7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb (expected
│ 6a2ee6c7b448462d81d4aa4b2d2bf0b25b1c1f3438c646e7e601f6e583bdfb18)
╵


❯ cat terraform-provider-libvirt_0.7.2_SHA256SUMS
094fb78b242783b2127faf926576abdf9c567b65f824a83c1d31214e2c2b3f56  terraform-provider-libvirt_0.7.2_windows_arm.zip
0d98f4d3cd38ba23eac10c5c7c35f2c5be155da8273dc5b042168841b28f2e48  terraform-provider-libvirt_0.7.2_freebsd_amd64.zip
1c96cdd3c556661963253938603d9ad5f38039b5af46b3236c3aa55b4995eaa6  terraform-provider-libvirt_0.7.2_linux_386.zip
1e56f930a36c714f4c3cdee81776e789e490d4c2edc388d4f9d97fcf002d1309  terraform-provider-libvirt_0.7.2_darwin_amd64.zip
293956c902f4196aa91b8c228672efcc975f0c2144d64f4fcfd3cbd0892262a2  terraform-provider-libvirt_0.7.2_windows_386.zip
3312206873dce42919b83b45c0e2892f4a802c46d0411aa8959c7ea2b1aefaf3  terraform-provider-libvirt_0.7.2_darwin_arm64.zip
50c6d51ff924ef815fc88dbb533cdca0742cf750368316cd77353c2924a8ed1a  terraform-provider-libvirt_0.7.2_windows_arm64.zip
542a9b7c49b6ade6252fb972e5975d8c0670cff126d1f75b9c86385974cd71ff  terraform-provider-libvirt_0.7.2_freebsd_arm.zip
6afcaa69b0e4dd360eab51b62baf80a33d780f97757eea256003b3bfeaaba6f5  terraform-provider-libvirt_0.7.2_freebsd_386.zip
7009cfe1e10ceaaf82ecd8425f3676e6241103ac455ccb863a9cafa4b86cd1bb  terraform-provider-libvirt_0.7.2_linux_arm64.zip
7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb  terraform-provider-libvirt_0.7.2_linux_amd64.zip
80b04c2546541c824e926838a128dca71bfed8c9d926f25bd49ebfebc00ad25f  terraform-provider-libvirt_0.7.2_freebsd_arm64.zip
a1238c15f307e1fdabfe1a7e1c9a4e0c1e6a1027a624c3e1bff236dc82a189a3  terraform-provider-libvirt_0.7.2_windows_amd64.zip
e7a1b04b3eab222f589f773a5c19810ea4497dabd4c3456efc6b7c271fe7e74b  terraform-provider-libvirt_0.7.2_linux_arm.zip
```

**Comment by bitchecker** [comment]
> │ Error: Failed to install provider
> │
> │ Error while installing dmacvicar/libvirt v0.7.2: checksum list has unexpected SHA-256 hash
> │ 7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb (expected
> │ 6a2ee6c7b448462d81d4aa4b2d2bf0b25b1c1f3438c646e7e601f6e583bdfb18)
> ╵

I'm experiencing the same issue just now:
```
╷
│ Error: Failed to install provider
│ 
│ Error while installing dmacvicar/libvirt v0.7.2: checksum list has unexpected SHA-256 hash 7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb (expected
│ 6a2ee6c7b448462d81d4aa4b2d2bf0b25b1c1f3438c646e7e601f6e583bdfb18)
╵

```

**Comment by JorgeAyllon** [comment]
El mismo error hoy 10-Oct-2023:
│ Error: Failed to install provider
│ 
│ Error while installing dmacvicar/libvirt v0.7.2: checksum list has unexpected SHA-256 hash
│ 7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb (expected
│ 6a2ee6c7b448462d81d4aa4b2d2bf0b25b1c1f3438c646e7e601f6e583bdfb18)

**Comment by krokwen** [comment]
> El mismo error hoy 10-Oct-2023: │ Error: Failed to install provider │ │ Error while installing dmacvicar/libvirt v0.7.2: checksum list has unexpected SHA-256 hash │ 7d634ac045a8593bdac82e3ae40589d81900262373caafbfa28c4d7c54bf91cb (expected │ 6a2ee6c7b448462d81d4aa4b2d2bf0b25b1c1f3438c646e7e601f6e583bdfb18)

I have same issue

**Comment by krokwen** [comment]
and now, on 0.7.3 the pgp key expired again

**Comment by dmacvicar** [comment]
Yes. I am doing some dummy releases while I figure it out. I am already signing with the extended key, but the registry code has a key of mine and does not allow to change it. I will try generating a whole new key.

**Comment by dmacvicar** [comment]
Ok pals, v0.7.4 should be working. Reopen if having problems.

**Comment by southsidedean** [comment]
Looks good with TF 1.6.1.
